grunt/helpers.js

```diff
@@ -43,21 +43,41 @@ module.exports = function(grunt) {
     // privates
     var generateIncludedRegExp = function() {
         var includes = grunt.config('includes') || [];
-        var re = '';
-        for(var i = 0, count = includes.length; i < count; i++) {
-            re += '\/' + includes[i].toLowerCase() + '\/';
-            if(i < includes.length-1) re += '|';
-        }
+        var pluginTypes = exports.defaults.pluginTypes;
+
+        // Return a more specific plugin regExp including src path.
+        var re = _.map(includes, function(plugin) {
+            return _.map(pluginTypes, function(type) {
+                return exports.defaults.sourcedir + type + '\/' + plugin + '\/';
+            }).join('|');
+        }).join('|');
+        return new RegExp(re, "i");
+    };
+
+    var generateNestedIncludedRegExp = function() {
+        var includes = grunt.config('includes') || [];
+        var pluginTypes = exports.defaults.pluginTypes;
+        var folderRegEx = /(\/less\/plugins)/;
+
+        // Return a more specific plugin regExp including src path.
+        var re = _.map(includes, function(plugin) {
+            return _.map(pluginTypes, function(type) {
+                return exports.defaults.sourcedir + type + '\/' + plugin + '\/' + folderRegEx;
```

Comment:
```jaavascript
exports.defaults.sourcedir + type + '\/' + plugin + '\/' + folderRegEx;
```
^ this line
```javascript
exports.defaults = "C:/path_to_files/";
type = "components" || "extensions" || "menu" || "theme";
plugin = "adapt-contrib-pluginName";
folderRegEx = /(/less/plugins)/;
```
folderRegExp evaluates to this string
```javascript
"\/(\/less/plugins)\/"
```
so the regex strings come out looking like this:
```javascript
"C:/path_to_files/components/adapt-contrib-pluginName//(/less/plugins)/"
```
when that parses a new regex, it's coming out like the following for every plugin:
```javascript
"C:/path_to_files/components/adapt-contrib-pluginName///less/plugins/"
```
to get rid of that triple slash we need folderRegExp to not be a regex parsed into a string and instead can be a string directly:
```javascript
folderRegEx = "less/plugins";
```
But more than than, we're trying to match the following inside every other plugin:
```javascript
"/less/plugin/adapt-contrib-pluginName/"
```
i think generateNestedIncludeRegExp needs to be:
```javscript
    var generateNestedIncludedRegExp = function() {
        var includes = grunt.config('includes') || [];
        var folderRegEx = "less/plugins/";

        // Return a more specific plugin regExp including src path.
        var re = _.map(includes, function(plugin) {
            return exports.defaults.sourcedir + '([^\/]*)\/([^\/]*)\/' + folderRegEx + plugin + '\/';
        }).join('|');
        return new RegExp(re, "i");
    };
```
which is the equivalent of trying to match:
```javascript
path_to_src/*/*/less/plugins/adapt-pluginName/
```

Reply:
Thanks @oliverfoster have pushed this change after testing it 👍